### PR TITLE
feat: group/user admin fixes and polish

### PR DESCRIPTION
**Pull request by echl**
# Description

* Implement service methods
* Loading skeletons/indicators where applicable
* Refresh users list when changing permissions in group and user view
* And and delete users to groups

https://github.com/DARPA-ASKEM/terarium/assets/16358728/61afa75f-9185-4201-a879-f86496cad280

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/permissions/GroupsController.java [hunk 1]
```diff
@@ -40,9 +40,9 @@ public ResponseEntity<List<PermissionGroup>> getGroups(
 		@RequestParam(name = "page", defaultValue = "0") Integer page
 	) {
 		List<PermissionGroup> groups = reBACService.getGroups();
-		groups.forEach(group -> {
-			group.setPermissionRelationships((new PermissionRelationships()));
-		});
+		// groups.forEach(group -> {
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/permissions/GroupsController.java:43]: may as well remove?

File: packages/client/hmi-client/src/services/groups.ts [hunk 2]
```diff
@@ -17,7 +17,10 @@ export async function addGroupUserPermissions(
 	relationship: string
 ) {
 	const response = await API.post(`/groups/${groupId}/permissions/user/${userId}/${relationship}`);
-	return response?.data ?? null;
+	if (response.status === 200) {
+		return true;
+	}
+	return false;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/groups.ts:23]:
```suggestion
	return response.status === 200;
```

Why the if/else?

File: packages/client/hmi-client/src/services/roles.ts
Comment (YohannParis) [inline packages/client/hmi-client/src/services/roles.ts:1]: Missing some try/catch and `logger` to notify the user about errors.

File: packages/client/hmi-client/src/services/user.ts
Comment (YohannParis) [inline packages/client/hmi-client/src/services/user.ts:1]: Same things about error handling

**Review by YohannParis — APPROVED** [review]

**Pull request merged**